lib/config-generator.js

```diff
@@ -107,7 +107,7 @@ class ConfigGenerator {
             config.resolve.alias['react-dom'] = 'preact-compat';
         }
 
-        config.externals = Object.assign({}, this.webpackConfig.externals);
+        config.externals = this.webpackConfig.externals;
```

Comment:
It would probably be better to copy the array here to avoid it from being modified between two `getWebpackConfig()` calls:

```suggestion
        config.externals = [...this.webpackConfig.externals];
```

It won't prevent deep changes (maybe we'll have to do something about this later) but will handle obvious "adding/removing an external" situations.

Reply:
I considered this, but figured since users can only add externals there would be no problem directly referencing the internal array. The only other indirect manipulation of this array is via the reset() function when it creates a whole new webpackConfig.

Reply:
That's a bit of an edge case but the issue is that the generated config can be manipulated at will after being generated.

For instance with a multi-config export:

```js
const config1 = Encore.getWebpackConfig();
const config2 = Encore.getWebpackConfig();

config1.externals.push('onlyConf1');

module.exports = [config1, config2];
```

Each call to `getWebpackConfig()` should generate its own config object... but if the array is not copied the `onlyConf1` external will be present in both of them.

As I said above, that's probably an edge case since most of the time people won't call `getWebpackConfig()` multiple times without using `reset()`... but it doesn't cost much to prevent it (well... at least the most obvious cases with a shallow copy).